net: dns-sd: allow empty instances

toonst opened the pull request:
when receiving DNS SD queries, the instance field can be empty, so they should be allowed.

With 
```
CONFIG_DNS_SD_LOG_LEVEL_DBG=y
```
enabled, and if you are connected to a local network with some mDNS activity, you can see that a lot of queries are unnecessarily dropped because they don't have an instance field:
```
[00:42:36.768,187] <dbg> net_dns_sd: instance_is_valid: (net_socket_service): instance is NULL
[00:42:36.768,192] <dbg> net_dns_sd: dns_sd_rec_match: DNS SD record at 0x10090810 is invalid
[00:42:36.768,215] <dbg> net_dns_sd: instance_is_valid: (net_socket_service): instance '' is too small (0, min: 1)
[00:42:36.768,231] <dbg> net_dns_sd: dns_sd_rec_match: DNS SD record at 0x180b5dec is invalid
```

toonst committed:
net: dns-sd: allow empty instances

when receiving DNS SD queries, the instance field can be empty, so they
should be allowed.

Signed-off-by: Toon Stegen <toon@toostsolutions.be>

cfriedt commented:
@toonst - thanks for looking into this.

In a separate commit (ideally ordered first), can you please also update tests so that they will pass with this change?

```shell
west build -p -b mps2/an385 tests/net/lib/dns_sd -T net.dns.sd.no_ipv6
```

toonst committed:
tests: net: dns-sd: allow for empty instances

it is valid to receive MDNS queries with empty instance name, in so
called "Service Enumeration Queries".

Signed-off-by: Toon Stegen <toon@toostsolutions.be>

jukkar commented on tests/net/lib/dns_sd/src/main.c:
```diff
@@ -304,16 +304,30 @@ ZTEST(dns_sd, test_add_ptr_record)
 				DNS_SD_EMPTY_TXT,
 				CONST_PORT);
-	zassert_equal(-EINVAL, add_ptr_record(&null_label, ttl,
+	zassert_equal(56, add_ptr_record(&null_instance, ttl,
```
Using magic values like 56 here is not good, it is very difficult for the next developer to understand what the value means. So please adjust this a bit, for example use a separate variable that has descriptive name for this, so something like
```
int record_offset;

record_offset = add_ptr_record(&null_instance, ttl, ...);
zassert_equal(record_offset, xxx, "some explanation here...");
```
or similar, the above is just a suggestion, please adjust when/if needed.

jukkar reviewed (commented):
Something is wrong with the tests. You can run the DNS SD tests locally like this using twister
```
west twister -c -O ../build/dns-sd-tests -T tests/net/lib/dns_sd
```
